diff --git a/setup.go b/setup.go
index 8b3c9bc..8eac97a 100644
--- a/setup.go
+++ b/setup.go
@@ -17,11 +17,11 @@
 package ads
 
 import (
+	"time"
+
 	"github.com/caddyserver/caddy"
 	"github.com/coredns/coredns/core/dnsserver"
 	"github.com/coredns/coredns/plugin"
-	"github.com/coredns/coredns/plugin/metrics"
-	"time"
 )
 
 const Version = "0.2.3"
@@ -56,8 +56,6 @@ func setup(c *caddy.Controller) error {
 
 	c.OnStartup(func() error {
 		once.Do(func() {
-			metrics.MustRegister(c, requestCount)
-			metrics.MustRegister(c, blockedRequestCount)
 			updater.Start()
 		})
 		return nil
