diff --git a/setup.go b/setup.go
index f20b325..2bcf226 100644
--- a/setup.go
+++ b/setup.go
@@ -4,7 +4,6 @@ import (
 	"github.com/caddyserver/caddy"
 	"github.com/coredns/coredns/core/dnsserver"
 	"github.com/coredns/coredns/plugin"
-	"github.com/coredns/coredns/plugin/metrics"
 )
 
 func init() { plugin.Register(pluginName, setup) }
@@ -17,21 +16,13 @@ func setup(c *caddy.Controller) error {
 		return PluginError(err)
 	}
 
-	r := &Dnsredir{ Upstreams: &ups }
+	r := &Dnsredir{Upstreams: &ups}
 	dnsserver.GetConfig(c).AddPlugin(func(next plugin.Handler) plugin.Handler {
 		r.Next = next
 		return r
 	})
 
 	c.OnStartup(func() error {
-		metrics.MustRegister(c,
-			NameLookupDuration,
-			RequestDuration,
-			RequestCount,
-			RcodeCount,
-			HealthCheckFailureCount,
-			HealthCheckAllDownCount,
-		)
 		return r.OnStartup()
 	})
 
@@ -41,4 +32,3 @@ func setup(c *caddy.Controller) error {
 
 	return nil
 }
-
