diff --git a/plugin/forward/forward.go b/plugin/forward/forward.go
index 897d9e7e..46badced 100644
--- a/plugin/forward/forward.go
+++ b/plugin/forward/forward.go
@@ -8,6 +8,7 @@ import (
 	"context"
 	"crypto/tls"
 	"errors"
+	"strings"
 	"time"
 
 	"github.com/coredns/coredns/plugin"
@@ -30,7 +31,7 @@ type Forward struct {
 	hcInterval time.Duration
 
 	from    string
-	ignored []string
+	ignored map[string]struct{}
 
 	tlsConfig     *tls.Config
 	tlsServerName string
@@ -44,7 +45,7 @@ type Forward struct {
 
 // New returns a new Forward.
 func New() *Forward {
-	f := &Forward{maxfails: 2, tlsConfig: new(tls.Config), expire: defaultExpire, p: new(policy.Random), from: ".", hcInterval: hcInterval}
+	f := &Forward{maxfails: 2, tlsConfig: new(tls.Config), expire: defaultExpire, p: new(policy.Random), from: ".", hcInterval: hcInterval, ignored: make(map[string]struct{})}
 	return f
 }
 
@@ -174,11 +175,14 @@ func (f *Forward) isAllowedDomain(name string) bool {
 		return true
 	}
 
-	for _, ignore := range f.ignored {
-		if plugin.Name(ignore).Matches(name) {
+	ss := strings.Split(name, ".")
+	for i := 0; i < len(ss); i++ {
+		n := strings.Join(ss[i:], ".")
+		if _, ok := f.ignored[n]; ok {
 			return false
 		}
 	}
+
 	return true
 }
 
diff --git a/plugin/forward/setup.go b/plugin/forward/setup.go
index 73c9cf94..1831b403 100644
--- a/plugin/forward/setup.go
+++ b/plugin/forward/setup.go
@@ -132,9 +132,8 @@ func parseBlock(c *caddy.Controller, f *Forward) error {
 			return c.ArgErr()
 		}
 		for i := 0; i < len(ignore); i++ {
-			ignore[i] = plugin.Host(ignore[i]).Normalize()
+			f.ignored[plugin.Host(ignore[i]).Normalize()] = struct{}{}
 		}
-		f.ignored = ignore
 	case "max_fails":
 		if !c.NextArg() {
 			return c.ArgErr()
