diff --git a/plugin/forward/forward.go b/plugin/forward/forward.go
index 624d0896..644dd42e 100644
--- a/plugin/forward/forward.go
+++ b/plugin/forward/forward.go
@@ -8,6 +8,7 @@ import (
 	"context"
 	"crypto/tls"
 	"errors"
+	"strings"
 	"sync/atomic"
 	"time"
 
@@ -34,7 +35,7 @@ type Forward struct {
 	hcInterval time.Duration
 
 	from    string
-	ignored []string
+	ignored map[string]struct{}
 
 	tlsConfig     *tls.Config
 	tlsServerName string
@@ -55,7 +56,7 @@ type Forward struct {
 
 // New returns a new Forward.
 func New() *Forward {
-	f := &Forward{maxfails: 2, tlsConfig: new(tls.Config), expire: defaultExpire, p: new(random), from: ".", hcInterval: hcInterval, opts: options{forceTCP: false, preferUDP: false, hcRecursionDesired: true}}
+	f := &Forward{maxfails: 2, tlsConfig: new(tls.Config), expire: defaultExpire, p: new(random), from: ".", hcInterval: hcInterval, opts: options{forceTCP: false, preferUDP: false, hcRecursionDesired: true}, ignored: make(map[string]struct{})}
 	return f
 }
 
@@ -201,11 +202,14 @@ func (f *Forward) isAllowedDomain(name string) bool {
 		return true
 	}
 
-	for _, ignore := range f.ignored {
-		if plugin.Name(ignore).Matches(name) {
+	ss := strings.Split(name, ".")
+	for i := 0; i < len(ss); i++ {
+		n := strings.Join(ss[i:], ".")
+		if _, ok := f.ignored[n]; ok {
 			return false
 		}
 	}
+
 	return true
 }
 
