diff --git a/plugin/forward/setup.go b/plugin/forward/setup.go
index 73c9cf94..1831b403 100644
--- a/plugin/forward/setup.go
+++ b/plugin/forward/setup.go
@@ -132,9 +132,8 @@ func parseBlock(c *caddy.Controller, f *Forward) error {
 			return c.ArgErr()
 		}
 		for i := 0; i < len(ignore); i++ {
-			ignore[i] = plugin.Host(ignore[i]).Normalize()
+			f.ignored[plugin.Host(ignore[i]).Normalize()] = struct{}{}
 		}
-		f.ignored = ignore
 	case "max_fails":
 		if !c.NextArg() {
 			return c.ArgErr()
